packages/salesforcedx-vscode-core/src/commands/util/sfdxCommandlet.ts

```diff
@@ -95,28 +111,55 @@ export abstract class SfdxCommandletExecutor<T>
     });
 
     execution.processExitSubject.subscribe(exitCode => {
-      const telemetryData = this.getTelemetryData(
-        exitCode === 0,
-        response,
-        output
-      );
-      let properties;
-      let measurements;
-      if (telemetryData) {
-        properties = telemetryData.properties;
-        measurements = telemetryData.measurements;
-      }
-      this.logMetric(
-        execution.command.logName,
-        startTime,
-        properties,
-        measurements
-      );
-      this.onDidFinishExecutionEventEmitter.fire(startTime);
+      this.exitProcessHandler(exitCode, execution, response, startTime, output);
     });
+
     this.attachExecution(execution, cancellationTokenSource, cancellationToken);
   }
 
+  protected exitProcessHandler(
+    exitCode: number | undefined,
+    execution: CliCommandExecution,
+    response: ContinueResponse<T>,
+    startTime: [number, number],
+    output: string
+  ): void {
+    if (
+      exitCode === 0 &&
+      execution.command.logName === FORCE_SOURCE_PULL_LOG_NAME
+    ) {
+      const pullResult = JSON.parse(output);
+      this.updateCache(pullResult);
+
+      const pullParser = new ForcePullResultParser(output);
+      const errors = pullParser.getErrors();
+      if (errors) {
+        channelService.showChannelOutput();
```

Comment:
Not sure about this part where errors are shown if encountered - need to generate or fake an error in the pull op to see how this looks.

Reply:
jest test FTW